op-e2e/op_geth_test.go

```diff
@@ -0,0 +1,268 @@
+	require.Nil(t, err)
+
+	// Finally create the engine client
+	client, err := sources.NewEngineClient(
+		l2Node,
+		log,
+		nil,
+		sources.EngineClientDefaultConfig(&rollup.Config{Genesis: rollupGenesis}),
+	)
+	require.Nil(t, err)
+
+	// Create the test data
+	// L1 Block Info Deposit
+	// Tx From GPO Owner (alice) to decrease the GPO params
+	// Tx to fund new account (enough ETH for the new account to pay for fees in the new but not old regime)
+	// Tx from the new account (should succeed in FCU, but fail in new payload).
+	// Note: It would be easy to go the other way (increase GPO params), but that would imply a block that fails
+	// in FCU but succeeds in NewPayload. I intend to test it, but it requires fully hardcoded data (to test that
+	// the NewPayload succeeds)
+	l1Info, err := derive.L1InfoDepositBytes(1, l1Block)
+	require.Nil(t, err)
+
+	gpoContract, err := bindings.NewGasPriceOracle(common.HexToAddress(predeploys.GasPriceOracle), l2Seq)
+	require.Nil(t, err)
+	l2opts, err := bind.NewKeyedTransactorWithChainID(cfg.Secrets.Alice, cfg.L2ChainIDBig())
+	require.Nil(t, err)
+
+	setScalarTx, err := gpoContract.SetScalar(l2opts, big.NewInt(0))
+	require.Nil(t, err)
+	setScalar, err := setScalarTx.MarshalBinary()
+	require.Nil(t, err)
+
+	rawFundTx := types.NewTransaction(1, cfg.Secrets.Addresses().Mallory, big.NewInt(50_000*params.GWei), 30000, big.NewInt(5*params.GWei), nil)
+	fundTx, err := l2opts.Signer(cfg.Secrets.Addresses().Alice, rawFundTx)
+	require.Nil(t, err)
+	fund, err := fundTx.MarshalBinary()
+	require.Nil(t, err)
+
+	rawInvalidTx := types.NewTransaction(0, cfg.Secrets.Addresses().Bob, big.NewInt(0), 25_000, big.NewInt(1*params.GWei), nil)
+	invalidTx, err := types.SignTx(rawInvalidTx, types.LatestSignerForChainID(cfg.L2ChainIDBig()), cfg.Secrets.Mallory)
+	require.Nil(t, err)
+	invalid, err := invalidTx.MarshalBinary()
+	require.Nil(t, err)
+
+	attrs := eth.PayloadAttributes{
+		Timestamp:    hexutil.Uint64(l2GenesisBlock.Time() + 2),
+		Transactions: []hexutil.Bytes{l1Info, setScalar, fund, invalid},
+		NoTxPool:     true,
+	}
+
+	// Go through the flow of FCU, GetPayload, NewPayload, FCU
+	// FCU should succeed, but NewPayload should fail.
+	ctx, cancel = context.WithTimeout(context.Background(), 2*time.Second)
+	defer cancel()
+
+	fc := eth.ForkchoiceState{
+		HeadBlockHash: l2GenesisBlock.Hash(),
+		SafeBlockHash: l2GenesisBlock.Hash(),
+	}
+	res, err := client.ForkchoiceUpdate(ctx, &fc, &attrs)
+	require.Nil(t, err)
+	require.Equal(t, eth.ExecutionValid, res.PayloadStatus.Status)
+	require.NotNil(t, res.PayloadID)
+
+	payload, err := client.GetPayload(ctx, *res.PayloadID)
+	require.Nil(t, err)
+	require.NotNil(t, payload)
+	require.Equal(t, payload.Transactions, attrs.Transactions) // Ensure we don't drop the transactions
+
+	status, err := client.NewPayload(ctx, payload)
+	require.Nil(t, err)
+	require.Equal(t, eth.ExecutionInvalid, status.Status)
+
+	// fc.HeadBlockHash = payload.BlockHash
```

Comment:
I'm guessing this needs a little more cleanup before its ready?

Reply:
So this line isn't strictly necessary, but is there for completeness. The part that does need to be updated when the op-geth PR is merged is the above line that changes from `ExecutionInvalid` to `ExecutionValid`.